### PR TITLE
feat: add Node Template support

### PR DESCRIPTION
**Pull request by zorrofox**
Add [node template](https://karpenter.sh/v0.27.0/concepts/node-templates/) support.
---
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by andskli — COMMENTED** [review]
@zorrofox this looks good to me. Did your integration tests pass successfully?

File: API.md
Comment (andskli) [inline API.md:1]: Actually, this should perhaps not be included as I think it's generated upon build time?

Comment (zorrofox) [reply API.md:1]: I have modify the integ testing for the [Node Template](https://github.com/zorrofox/cdk-eks-karpenter/blob/24bff8b0e90e484fa1f96e1d84d87d2edef8d9b9/test/integ.karpenter.ts#L53).

**Comment by andskli** [comment]
@zorrofox what I am saying is that I do not think that API.md should be committed to this repository as it is auto-generated upon build.

**Comment by zorrofox** [comment]
> 

You can see, if I not include the `API.md`, the Gihub Action will run into failed.

**Comment by andskli** [comment]
@zorrofox I see, let's add it back in then

**Pull request merged**